app/src/main/java/io/github/hidroh/materialistic/WebFragment.java

```diff
@@ -307,12 +308,26 @@ private void reloadUrl(String url) {
 
     @SuppressLint("AddJavascriptInterface")
     private void reloadUrl(String url, @Nullable String pdfFilePath) {
+        mIsPdf = false;
         if (mPdfAndroidJavascriptBridge != null) {
             mPdfAndroidJavascriptBridge.cleanUp();
             mWebView.removeJavascriptInterface("PdfAndroidJavascriptBridge");
         }
         if (pdfFilePath != null && isPdfRenderingSupported() && TextUtils.equals(PDF_LOADER_URL, url)) {
-            mPdfAndroidJavascriptBridge = new PdfAndroidJavascriptBridge(pdfFilePath, this::offerExternalApp);
+            mProgressBar.setProgress(80);
+            mIsPdf = true;
+            mPdfAndroidJavascriptBridge = new PdfAndroidJavascriptBridge(pdfFilePath, new PdfAndroidJavascriptBridge.Callbacks() {
+                @Override
+                public void onFailure() {
+                    offerExternalApp();
+                }
+
+                @Override
+                public void onLoad() {
+                    mProgressBar.setProgress(100);
+                    mProgressBar.setVisibility(GONE);
```

Comment:
Maybe it would be more consistent if we extract the logic from `onProgressChanged()` and call it instead? That method handles progress bar visibility and refresh button icon. Same goes for when we set progress to 80.